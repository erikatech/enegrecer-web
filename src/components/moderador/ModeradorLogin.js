import React, { Component } from 'react';
import { auth, storageKey } from '../../utils/firebaseUtils';
import Modal from '../comum/modal/modal';
import './moderador-login.css';

export default class ModeradorLogin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      senha: '',
      error: null,
      deveExibirPopup: false
    };
  }

  componentDidMount() {
    // auth.onAuthStateChanged((user) => {
    //   if (user.uid.includes('obr3tOVUmgfM0WYf51ftqE4zuqz2')) {
    //     window.localStorage.setItem(storageKey, user.uid);
    //   } else {
    //     window.localStorage.removeItem(storageKey);
    //   }
    // });
  }

  onSubmit = (event) => {
    const {
      email,
      senha,
    } = this.state;

    const self = this;

    auth.signInWithEmailAndPassword(email, senha).then(() => {
      window.location.href = '/moderador/painel';
    }).catch((error) => {
      const message = this.trataMensagemDeErro(error);
      self.setState({ deveExibirPopup: true, message });
    });

    event.preventDefault();
  }

  trataMensagemDeErro = (error) => {
    const errorMap = {
      'auth/wrong-password': 'Senha errada',
      'auth/user-not-found': 'Usuário não encontrado',
      'auth/user-disabled': 'Usuário inativo',
      'auth/invalid-email': 'Email inválido',
    };

    return errorMap[error.code];
  }

  render() {
    const {
      email,
      senha,
      error,
      deveExibirPopup
    } = this.state;

    const isInvalid =
      senha === '' ||
      email === '';

    return (
      <div className="row login_moderador">
        <form className="col s4 formulario_login_moderador" onSubmit={this.onSubmit}>
          <div className="input-field col s12">
            <input id="login_moderador" type="text" className="validate" value={email} onChange={event => this.setState({ email: event.target.value })} />
            <label htmlFor="login_moderador">Login Moderador</label>
          </div>
          <div className="input-field col s12">
            <input id="senha_moderador" type="password" className="validate" value={senha} onChange={event => this.setState({ senha: event.target.value })} />
            <label htmlFor="senha_moderador">Senha Moderador</label>
          </div>
          <button id="confirm-button" data-target="modal_erro" className="waves-effect waves-light btn" disabled={isInvalid} type="submit" name="action" >
              Entrar
          </button>
        </form>
        <Modal
          id="modal_erro"
          tituloModal="Erro ao logar"
          textoModal={error}
          textoBotao="FECHAR"
          acaoBotaoOk={() => console.log('ok button')}
          deveExibir={deveExibirPopup}
        />
      </div>
    );
  }
}
