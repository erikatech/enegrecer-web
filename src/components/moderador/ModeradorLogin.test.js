import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ModeradorLogin from './ModeradorLogin';

describe('ModeradorLogin', () => {
  it.only('deve construir o componente corretamente', () => {
    const moderadorLogin = shallow(<ModeradorLogin />);
    const moderadorLoginJson = toJson(moderadorLogin);
    expect(moderadorLoginJson).toMatchSnapshot();
  });

  it.only('deve mostrar o modal quando submeter o form com dados invÃ¡lidos', () => {
    const moderadorLoginWrapper = mount(<ModeradorLogin />);
    moderadorLoginWrapper.debug();

    const found = moderadorLoginWrapper.find('.modal');
    expect(found.exists()).toBeFalsy();
  });
});
