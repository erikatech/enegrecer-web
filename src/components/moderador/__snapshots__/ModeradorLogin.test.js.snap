// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModeradorLogin deve construir o componente corretamente 1`] = `
<div
  className="row login_moderador"
>
  <form
    className="col s4 formulario_login_moderador"
    onSubmit={[Function]}
  >
    <div
      className="input-field col s12"
    >
      <input
        className="validate"
        id="login_moderador"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label
        htmlFor="login_moderador"
      >
        Login Moderador
      </label>
    </div>
    <div
      className="input-field col s12"
    >
      <input
        className="validate"
        id="senha_moderador"
        onChange={[Function]}
        type="password"
        value=""
      />
      <label
        htmlFor="senha_moderador"
      >
        Senha Moderador
      </label>
    </div>
    <button
      className="waves-effect waves-light btn"
      data-target="modal_erro"
      disabled={true}
      id="confirm-button"
      name="action"
      type="submit"
    >
      Entrar
    </button>
  </form>
  <Modal
    acaoBotaoOk={[Function]}
    deveExibir={false}
    id="modal_erro"
    textoBotao="FECHAR"
    textoModal={null}
    tituloModal="Erro ao logar"
  />
</div>
`;
